### PR TITLE
Transfer Gateway/DPoSV2 Config

### PR DESCRIPTION
**Pull request by ajhodges**
This PR primarily addresses the need to include a loom.yaml config file with each node (#16). Additionally, the first node in a newly provisioned network will be designated as the transfer gateway oracle. The transfer gateway oracle _must_ be provisioned with an Ethereum private key that matches the address designated as the oracle in the mainnet gateway contract. Once the network is up, every initially provisioned node is whitelisted by the (DPoS) oracle and registered as a validator.

Other changes include:

- Addition of a Route53 DNS entry - {network-name}.network.shipchain.io
- Addition of an application load balancer pointing to all of the initial shipchain nodes for use in the dashboard, or any dapps.
- Configurable parameters for loom, transfer gateway, and DPoS
- ICMP traffic is opened to the nodes, addressing #15 
- Security groups/network ACLs cleaned up
- Lint issues
- Light refactoring

### CODE REVIEW

**Comment by ajhodges** [comment]
Network provisioning with these changes looks something like this:
```
hydra network provision --name hydra-testing -s 3 -v 2.0.0+b963
hydra network publish --name hydra-testing --version 2.0.0+b963
hydra network configure --name hydra-testing --oracle-priv-key oracle_eth_priv.key
```
with `oracle_eth_priv.key` being the Ethereum private key mentioned above.

File: hydra/helpers/client.py [hunk 1]
```diff
@@ -186,14 +213,52 @@ def configure(self, name, destination, **kwargs):
             'chaindata/config/genesis.json'
         )
 
+        # LOOM.YAML
+        self._copy_yaml(
+            f'{self.app.config["hydra"]["channel_url"]}/networks/{name}/loom.yaml',
+            'loom.yaml'
+        )
+        if kwargs['oracle'] if 'oracle' in kwargs else False:
```
Comment (mlclay) [inline hydra/helpers/client.py:251]: For the sake of keeping this growing method easier to read, can we extract this Oracle loom.yaml setup to its own method `_setup_oracle_loom_yaml`?

**Review by mlclay — CHANGES_REQUESTED** [review]
Worked with Adam testing these changes over the last week with different provisioned networks.  I like the changes and know they work well.  Minor refactor for readability.

**Comment by leewardbound** [comment]
W/r/t the Oracle key note -- I'm aware that the genesis.json needs a specific Oracle key, but when I asked about it, Loom replied that actually, every node is a TG oracle, not just 1 of them, and they behave as a multisig on the mainnet TG contract.

I assume the singular key that the genesis.json file requires is just to bootstrap the other validators, perhaps, adding them to the list of multisig addresses on the mainnet contract; but in the name of science, we need to consult w/ their team and figure out what relevance setting the initial oracle key actually has, and what-if-anything other nodes need to do w/r/t TG oracle keys.

Other than that, these changes look fantastic, very impressive work done ahead of schedule, thanks!

**Review by leewardbound — APPROVED** [review]

**Review by mlclay — APPROVED** [review]

**Pull request merged**